### PR TITLE
bots: Improve virtual_fs bot coverage

### PR DESCRIPTION
**Pull request by abhijeetkaur**
Used sample_conversation and other commands to improve code coverage from 54% to 94%

### CODE REVIEW

**Comment by derAnfaenger** [comment]
Seems to not work for Python 3

File: zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py [hunk 1]
```diff
@@ -24,6 +25,55 @@ class TestVirtualFsBot(BotTestCase):
...
+                                'read /yo/apple\nred\n\n'
+                                'mkdir /yo/apple\nERROR: file already exists\n\n'
+                                'ls /invalid\nERROR: file does not exist\n\n'
+                                'ls /foo\nERROR: /foo is not a directory\n\n'
+                                'ls /\n* /*bar*\n* /*foo*\n* /yo/\n\n'
+                                'invalid command\nERROR: unrecognized command\n\n'
+                                'write\nERROR: syntax: write <path> <some_text>\n\n'
+                                'help' + get_help() + '\n\n'
+                                'help ls\nsyntax: ls <optional_path>\n\n'
+                                'help invalid_command' + get_help() + '\n\n')
```
Comment (derAnfaenger) [inline zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py]: Can we just import this sample conversation from virtual_fs?

File: zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py [hunk 2]
```diff
@@ -46,3 +97,59 @@ def test_commands_2(self):
...
+            ('cd /yo/apple', 'foo_sender@zulip.com:\nERROR: /yo/*apple* is a file, directory required'),
+            ('mkdir /yo/apple', 'foo_sender@zulip.com:\nERROR: file already exists'),
+            ('ls /invalid', 'foo_sender@zulip.com:\nERROR: file does not exist'),
+            ('ls /foo', 'foo_sender@zulip.com:\nERROR: /foo is not a directory'),
+            ('ls /', 'foo_sender@zulip.com:\n* /*bar*\n* /*foo*\n* /yo/'),
+            ('invalid command', 'foo_sender@zulip.com:\nERROR: unrecognized command'),
+            ('write', 'foo_sender@zulip.com:\nERROR: syntax: write <path> <some_text>'),
+            ('help', self.help_txt),
+            ('help ls', 'foo_sender@zulip.com:\nsyntax: ls <optional_path>'),
+            ('help invalid_command', self.help_txt),
```
Comment (derAnfaenger) [inline zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py]: Same as above.

File: zulip_bots/zulip_bots/bots/virtual_fs/virtual_fs.py [hunk 3]
```diff
@@ -61,56 +61,54 @@ def get_help():
...
+            'read /yo/apple\nred\n\n'
+            'mkdir /yo/apple\nERROR: file already exists\n\n'
+            'ls /invalid\nERROR: file does not exist\n\n'
+            'ls /foo\nERROR: /foo is not a directory\n\n'
+            'ls /\n* /*bar*\n* /*foo*\n* /yo/\n\n'
+            'invalid command\nERROR: unrecognized command\n\n'
+            'write\nERROR: syntax: write <path> <some_text>\n\n'
+            'help' + get_help() + '\n\n'
+            'help ls\nsyntax: ls <optional_path>\n\n'
+            'help invalid_command' + get_help() + '\n\n')
```
Comment (derAnfaenger) [inline zulip_bots/zulip_bots/bots/virtual_fs/virtual_fs.py]: What's this change for? Shouldn't it work in a list of tuples as well?

Comment (derAnfaenger) [reply zulip_bots/zulip_bots/bots/virtual_fs/virtual_fs.py]: Or actually just a list is probably what we want here, but this is just one super-long string :)

File: zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py (hunk 1 above)
Comment (showell) [reply zulip_bots/zulip_bots/bots/virtual_fs/test_virtual_fs.py]: I think we really want to reuse the sample conversation from the bot itself, even if we have to post-process it a bit, so I agree with @roberthoenig here. :smile:

**Comment by showell** [comment]
@abhijeetkaur Are you still working on this?

**Comment by showell** [comment]
Hi, @abhijeetkaur, thanks for working on this!

I am closing this PR, as I think the following PR may have covered most of the same issues:

https://github.com/zulip/python-zulip-api/pull/295

When your time frees up again, there are probably still some holes in our coverage, so we may want to look at this again, but for now it's low priority.

**Pull request closed without merging**